uptane/services/director.py

```diff
@@ -380,6 +380,26 @@ def validate_primary_certification_in_vehicle_manifest(
 
     ecu_public_key = inventory.ecu_public_keys[primary_ecu_serial]
 
+    # Here, we check to see if the key that signed the Vehicle Manifest is the
+    # same key as ecu_public_key (the one the director expects), so that we can
+    # generate a more informative error, allowing user/debugger to distinguish
+    # between a bad signature ostensibly from the right key and a signature
+    # from the wrong key.
+    # TODO: Fix(?) assumption that one signature is used below.
+    keyid_used_in_signature = vehicle_manifest['signatures'][0]['keyid']
+    # Note, though, that there could be some edge cases here that the TUF code
+    # might actually resolve: for example, if the keyid hash algorithm used
+    # in the signature is not the same one as the one used in the key listing,
+    # this check would provide a false failure. So we don't raise an error here,
+    # and instead just log this difference and let the final arbiter of the
+    # validity of the signature be the dedicated code in tuf.keys.
+    if keyid_used_in_signature != ecu_public_key['keyid']:
+      log.info(
+          'Key used to sign Vehicle Manifest has a different keyid from that '
+          'listed in the inventory DB. Expect signature validation to fail, '
+          'unless the key is the same but the keyid differently hashed.')
```

Comment:
I think it will assist debuggers if the 'keyid_used_in_signature' and the expected keyid are included in the logged message.